src/worker.ts

```diff
@@ -95,9 +98,13 @@ export class Worker extends EventEmitter {
     this.concurrency = options.concurrency || 20;
     this.shutdownTimeout = (options.timeout || 8) * 1000;
     this.beatInterval = (options.beatInterval || 15) * 1000;
-    this.queues = options.queues || [];
-    if (this.queues.length === 0) {
-      this.queues = ["default"];
+    const queues = options.queues || ["default"];
+    if (typeof queues === "function") {
```

Comment:
@ThePletch

This PR also supports providing a function that returns queues. I haven't documented it publicly, but its there if you need it.